configuration/cdap-site.xml

```diff
@@ -1478,6 +1490,470 @@
+      <property>
+        <type>cdap-site</type>
+        <name>security.authentication.handlerClassName</type>
+      </property>
+    </depends-on>
+  </property>
+
+  <property>
+    <name>security.authentication.handler.ldapsVerifyCertificate</name>
+    <value>true</value>
+    <description>
+      Enable verification of the SSL certificate used by the LDAP server (LDAP only)
+    </description>
+    <depends-on>
+      <property>
+        <type>cdap-site</type>
+        <name>security.authentication.handlerClassName</type>
+      </property>
+    </depends-on>
+  </property>
+
+  <property>
+    <name>security.authentication.loginmodule.className</name>
+    <value></value>
+    <description>
+      JAAS LoginModule implementation to use when
+      co.cask.security.server.JAASAuthenticationHandler is configured for
+      security.authentication.handlerClassName
+    </description>
+    <depends-on>
+      <property>
+        <type>cdap-site</type>
+        <name>security.authentication.handlerClassName</type>
+      </property>
+    </depends-on>
+    <value-attributes>
+      <empty-value-valid>true</empty-value-valid>
+    </value-attributes>
+  </property>
+
+  <property>
+    <name>security.authorization.enabled</name>
+    <value>false</value>
+    <description>
+      When set to true, all operations in CDAP are authorized using the authorizer implementation found at the
+      property security.authorization.extension.jar.path.
+    </description>
+    <display-name>CDAP Authorization Extension Enabled</display-name>
+    <depends-on>
+      <property>
+        <type>cdap-site</type>
+        <name>security.authentication.extension.jar.path</type>
+      </property>
+    </depends-on>
+    <value-attributes>
+      <type>boolean</type>
+      <overridable>false</overridable>
+    </value-attributes>
+    <depends-on>
+      <property>
+        <type>cdap-site</type>
+        <name>security.enabled</type>
+      </property>
+    </depends-on>
+  </property>
+
+  <property>
+    <name>security.authorization.extension.jar.path</name>
+    <value></value>
+    <description>
+      If an external authorization system is used for authorizing operations on CDAP entities, this property sets
+      the path to the bundled JAR file containing the extension code. This jar is only used when
+      authorization is enabled by setting security.authorization.enabled to true.
+    </description>
+  </property>
+
+  <property>
+    <name>security.enabled</name>
+    <value>false</value>
+    <description>
+      Determines if authentication is enabled for CDAP; if true, all
+      requests to CDAP must provide a valid access token
+    </description>
+    <display-name>CDAP Authentication Enabled</display-name>
+    <value-attributes>
+      <type>boolean</type>
+      <overridable>false</overridable>
+    </value-attributes>
+    <depends-on>
+      <property>
+        <type>cdap-site</type>
+        <name>kerberos.auth.enabled</type>
```

Comment:
Correct. That's what our documentation says. We store keys in ZooKeeper and they would be readable by anyone, making auth moot, without kerberos. We allow it when testing only to simplify testing of auth independent of kerberos.